runtime/sema/type_tags.go

```diff
@@ -870,3 +877,47 @@ func commonSuperTypeOfComposites(types []Type) Type {
 
 	return superType
 }
+
+func unwrapOptionals(types []Type) ([]Type, int) {
+	unwrappedTypes := make([]Type, 0, len(types))
+
+	maxLevels := 0
+	for _, typ := range types {
+		levels := 0
+
+		// Unwrap optionals
+		for {
+			optionalType, ok := typ.(*OptionalType)
+			if !ok {
+				break
+			}
+
+			typ = optionalType.Type
+			levels++
+		}
+
+		maxLevels = max(maxLevels, levels)
+
+		unwrappedTypes = append(unwrappedTypes, typ)
+	}
+
+	return unwrappedTypes, maxLevels
+}
+
+func wrapOptionals(typ Type, levels int) Type {
+	for i := 0; i < levels; i++ {
+		typ = &OptionalType{
+			Type: typ,
+		}
+	}
+
+	return typ
+}
+
+func max(a, b int) int {
```

Comment:
Does this not already exist in the standard library?

Reply:
stdlibs doesn't have a one for `int`. That one takes `float64` and its implementation is a bit heavy